homeassistant/components/hdmi_cec/__init__.py

```diff
@@ -378,6 +378,26 @@ def __init__(self, device, logical) -> None:
         self._state: str | None = None
         self._logical_address = logical
         self.entity_id = "%s.%d" % (DOMAIN, self._logical_address)
+        self._attr_should_poll = False
+        self._attr_name = (
+            f"{self.vendor_name} {self._device.osd_name}"
+            if (
+                self._device.osd_name is not None
+                and self.vendor_name is not None
+                and self.vendor_name != "Unknown"
+            )
+            else "%s %d" % (self._device.type_name, self._logical_address)
+            if self._device.osd_name is None
+            else "%s %d (%s)"
+            % (self._device.type_name, self._logical_address, self._device.osd_name)
+        )
+        self._attr_icon = (
+            self._icon
+            if self._icon is not None
```

Comment:
The icon can now no longer be changed by setting `self._icon` after the entity has been initialized.
I can't find where `self._icon` is set. 😕 ❓

Reply:
self._icon is set depending on the type stored in self._device, and that is only set in _init.

Once __init is passed _icon and _device are not changed.